File: workspace/1_FindMaxValueInList/test/listutils/FindMaxValueInListTest.java
package listutils;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;

public class FindMaxValueInListTest {
	
	@Test(expected=IllegalArgumentException.class)
	public void emptyListThrowsException() {
		ListUtil.findMaxValueInList();
	}
	
	@Test
	public void listWithOneElement() {
		assertEquals(new Integer(2), ListUtil.findMaxValueInList(2));
	}
	
	@Test
	public void listWithMoreElementsMaxValueFirst() {
		assertEquals(new Integer(2), ListUtil.findMaxValueInList(2, 1, 0, 1));
	}
	
	@Test
	public void listWithMoreElementsMaxValueInTheMiddle() {
		assertEquals(new Integer(2), ListUtil.findMaxValueInList(1, 1, 0, 1, 2, 1));
	}
	
	@Test
	public void listWithMoreElementsMaxValueLast() {
		assertEquals(new Integer(2), ListUtil.findMaxValueInList(1, 1, 1, 1, 0, 2));
	}
	

}

File: solution/1_FindMaxValueInList/1_FindMaxValueInList/FindMaxValueInListTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using listutils;

namespace test
{
    [TestClass]
    public class FindMaxValueInListTest
    {
        [TestMethod]
        [ExpectedException(typeof(ArgumentException))]
	    public void emptyListThrowsException() {
		    ListUtil.findMaxValueInList();
	    }
	
	    [TestMethod]
	    public void listWithOneElement() {
		    Assert.AreEqual(2, ListUtil.findMaxValueInList(2));
	    }
	
	    [TestMethod]
	    public void listWithMoreElementsMaxValueFirst() {
		    Assert.AreEqual(2, ListUtil.findMaxValueInList(2, 1, 0, 1));
	    }
	
	    [TestMethod]
	    public void listWithMoreElementsMaxValueInTheMiddle() {
		    Assert.AreEqual(2, ListUtil.findMaxValueInList(1, 1, 0, 1, 2, 1));
	    }
	
	    [TestMethod]
	    public void listWithMoreElementsMaxValueLast() {
		    Assert.AreEqual(2, ListUtil.findMaxValueInList(1, 1, 1, 1, 0, 2));
	    }
    }
}
